pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1808,3 +1996,151 @@ func getSidecarInboundBindIP(node *model.Proxy) string {
+						node:            pluginParams.Node,
+						listenerName:    listenerMapKey,
+						currentServices: currentListenerEntry.services,
+						currentProtocol: currentListenerEntry.servicePort.Protocol,
+						newHostname:     newHostname,
+						newProtocol:     pluginParams.Port.Protocol,
+					}.addMetric(node, pluginParams.Push)
+					break compareWithExisting
+				} else {
+					continue
+				}
+			}
+			if incomingFilterChain.FilterChainMatch == nil {
+				continue
+			}
+
+			// We have two non-catch all filter chains. Check for duplicates
+			if reflect.DeepEqual(*existingFilterChain.FilterChainMatch, *incomingFilterChain.FilterChainMatch) {
+				var newHostname config.Hostname
+				if pluginParams.Service != nil {
+					newHostname = pluginParams.Service.Hostname
+				} else {
+					// user defined outbound listener via sidecar API
+					newHostname = "sidecar-config-egress-tcp-listener"
+				}
+
+				conflictFound = true
+				outboundListenerConflict{
+					metric:          model.ProxyStatusConflictOutboundListenerTCPOverTCP,
+					node:            pluginParams.Node,
+					listenerName:    listenerMapKey,
+					currentServices: currentListenerEntry.services,
+					currentProtocol: currentListenerEntry.servicePort.Protocol,
+					newHostname:     newHostname,
+					newProtocol:     pluginParams.Port.Protocol,
+				}.addMetric(node, pluginParams.Push)
+				break compareWithExisting
+			}
+		}
+
+		if !conflictFound {
+			// There is no conflict with any filter chain in the existing listener.
+			// So append the new filter chains to the existing listener's filter chains
+			newFilterChains = append(newFilterChains, incomingFilterChain)
+			if pluginParams.Service != nil {
+				lEntry := listenerMap[listenerMapKey]
+				lEntry.services = append(lEntry.services, pluginParams.Service)
+			}
+		}
+	}
+
+	return newFilterChains
+}
+
+func mergeFilterChains(httpFilterChain, tcpFilterChain []*listener.FilterChain) []*listener.FilterChain {
+	var newFilterChan []*listener.FilterChain
+	for _, fc := range httpFilterChain {
+		if fc.FilterChainMatch == nil {
+			fc.FilterChainMatch = &listener.FilterChainMatch{}
+		}
+
+		fc.FilterChainMatch.ApplicationProtocols = append(fc.FilterChainMatch.ApplicationProtocols, []string{"h2", "http/1.1", "http/1.0"}...)
+		newFilterChan = append(newFilterChan, fc)
+
+	}
+	return append(newFilterChan, tcpFilterChain...)
+}
+
+func getPluginFilterChain(opts buildListenerOpts) []plugin.FilterChain {
+	filterChain := make([]plugin.FilterChain, len(opts.filterChainOpts))
+
+	for id := range filterChain {
+		if opts.filterChainOpts[id].httpOpts == nil {
+			filterChain[id].ListenerProtocol = plugin.ListenerProtocolTCP
+		} else {
+			filterChain[id].ListenerProtocol = plugin.ListenerProtocolHTTP
+		}
+	}
+
+	return filterChain
+}
+
+func checkWellKnownPorts(port int, protocol protocol.Instance, conflict int) bool {
+	p, has := wellKnownPorts[port]
+	if conflict == 0 || !has || p != protocol {
```

Comment:
If you change `conflict == 0` to be the conflict type (XoverY, or whatever it is) it may be more clear

Reply:
Changed to `NoConflict`